src/vmcu/services/communications/Decoder_Data.py

```diff
@@ -0,0 +1,151 @@
+                bytes_count = 0
+                bytes_size = 0
+                wait_new_packet = True  
+
+
+            
+                
+    def deserialize(self, buffer, id_packet):
+
+        for type, name in zip(self.dict_measurement_types[id_packet], self.dict_measurement_names[id_packet]):
+            data = None
+            match type:
+                case "bool":
+                    data = 'True' if struct.unpack('<B', buffer[:1])[0] else 'False'
+                    buffer = buffer[1:]
+                case "uint8":
+                    data = struct.unpack('<B', buffer[:1])[0]
+                    buffer = buffer[1:]
+                case "int8":
+                    data = struct.unpack('<b', buffer[:1])[0]
+                    buffer = buffer[1:]
+                case "uint16":
+                    data = struct.unpack('<H', buffer[:2])[0]
+                    buffer = buffer[2:]
+                case "int16":
+                    data = struct.unpack('<h', buffer[:2])[0]
+                    buffer = buffer[2:]
+                case "uint32":
+                    data = struct.unpack('<I', buffer[:4])[0]
+                    buffer = buffer[2:]
+                case "int32":
+                    data = struct.unpack('<i', buffer[:4])[0]
+                    buffer = buffer[4:]
+                case "float32":
+                    data = struct.unpack('<f', buffer[:4])[0]
+                    buffer = buffer[4:]
+                case "uint64":
+                    data = struct.unpack('<Q', buffer[:8])[0]
+                    buffer = buffer[8:]
+                case "int64":
+                    data = struct.unpack('<q', buffer[:8])[0]
+                    buffer = buffer[8:]
+                case "float64":
+                    data = struct.unpack('<d', buffer[:8])[0]
+                    buffer = buffer[8:]
+                case _:
+                    if 'enum' in type:
+                        value_data = struct.unpack('<B', buffer[:1])[0]
+                        buffer = buffer[1:]
+                        valores = (re.search(r'\((.*?)\)', type)).group(1).split('-')
+                        data = valores[value_data]
+                        
+                    else:
+                        raise ValueError(f"Unsupported data type: {type}")
+            
+            self.dict_measurement_value[name] = data
+            
+
+    def calculate_byte_size(self, measurements):
+
+        size = 0
+        for measure in measurements:
+            if measure == 'uint8' or measure == 'int8' or measure == 'bool' or 'enum' in measure:
+                size += 1
+            elif measure == 'uint16' or measure == 'int16':
+                size += 2
+            elif measure == 'uint32' or measure == 'int32' or measure == 'float32':
+                size += 4
+            elif measure == 'uint64' or measure == 'int64' or measure == 'float64':
+                size += 8
+              
+        return size
```

Comment:
precalculate this in the constructor, the decoding function will be called a lot of times and we want to reduce the time it takes to complete